app/plugins/adp-backend/src/providers/deliveryProgrammeTransformers.test.ts

```diff
@@ -0,0 +1,45 @@
+import { defaultProgrammeGroupTransformer } from './deliveryProgrammeTransformers';
+
+describe('defaultProgrammeGroupTransformer', () => {
+  it('should transform valid ArmsLengthBody to GroupEntity', async () => {
+    const deliveryProgramme = {
+      programme_managers: [],
+      title: 'Test title 1',
+      alias: 'Test Alias',
+      description: 'Test description',
+      finance_code: 'Test finance_code',
+      arms_length_body: '05aa36b6-c7a2-4c35-820e-c31d20011f47',
+      delivery_programme_code: 'Test delivery_programme_code',
+      url: 'https://www.example.uk/',
+      name: 'test-title-1',
+      id: '1234',
+      created_at: new Date(),
+      updated_at: new Date(),
+    };
+
+    const expectedGroupEntity = {
+      apiVersion: 'backstage.io/v1beta1',
+      kind: 'Group',
+      metadata: {
+        name: 'test-title-1',
+        title: 'Test title 1 (Test Alias)',
+        description: 'Test description',
+        tags: [],
+        annotations: {
+          'backstage.io/managed-by-location':
+            'adp:delivery-programme\\test-title-1',
+          'backstage.io/managed-by-origin-location':
+            '`adp:delivery-programme\\${deliveryProgramme.name}`',
+        },
+        links: [{ url: 'https://www.example.uk/' }],
+      },
+      spec: {
+        type: 'delivery-programme',
```

Comment:
Now that we have the delivery programmes, the ALB catalog transformer logic should pull these in as children.

Reply:
@kbabigumira From what I recall from our call on Thursday this will be done in a new PR as we need to make changes to the entityprovider file and implement the tests for it too?

Reply:
@jamescoxheadcapgemini Sounds good thank you